infer/src/backend/InferAnalyze.ml

```diff
@@ -261,6 +261,7 @@ let main ~changed_files =
   let source_files = lazy (get_source_files_to_analyze ~changed_files) in
   (* empty all caches to minimize the process heap to have less work to do when forking *)
   clear_caches () ;
+  if Config.cil_lang then Language.curr_language := Language.CIL ;
```

Comment:
Is this needed? I would think that this is set at the beginning of the analysis of each procedure anyway so we'll get the correct language by the time we start analysing any code. Then we wouldn't need the new config option `Config.cil_lang`.

Reply:
Like I mentioned in the response to another comment, when we split `infer run` into `infer capture` and `infer analyze`, set `Language.curr_language` to `CIL` in  `CaptureSILJson` won't be fine for us because the language set in `infer capture` won't be passed to `infer analyze`. That is the reason why we ended up creating a `--cli-lang` flag. Do you have any better suggestion for this?